lib/spack/spack/test/stage.py

```diff
@@ -347,6 +347,24 @@ def create_stage_archive(expected_file_list=[_include_readme]):
     return create_stage_archive
 
 
+@pytest.fixture
```

Comment:
I generalized `create_stage_archive` to allow it to create an "expanding" archive when `expected_file_list` includes more than one "standard" file (e.g., `_extra_fn` and, optionally, `_hidden_fn`).

It is being tested by `test_expand_archive`.

Or am I missing something here?

Reply:
Whoops this does appear to be tested, so this (and the changes to handling of exploding tarballs in `fetch_strategy`) can be reverted - thanks!